os/arch/arm/src/common/up_createstack.c

```diff
@@ -164,6 +164,15 @@ int up_create_stack(FAR struct tcb_s *tcb, size_t stack_size, uint8_t ttype)
 {
 	size_t stack_alloc_size = stack_size;
 
+	/* new thread does not have the uheap value yet, It's added in the task_
+	 * schedsetup function.
+	 * Parent and child threads must have the same uheap value, so we are
+	 * using the parent thread uheap
+	 */
+#if defined(HAVE_KERNEL_HEAP) || defined(CONFIG_MPU_STACK_OVERFLOW_PROTECTION)
+	uint32_t uheap = sched_self()->uheap;	/* User heap pointer */
```

Comment:
How about this for readability?
```c
bool is_user_thread = (sched_self()->uheap != NULL);
```
I don't know above is right syntax or not.

or
```c
bool is_user_thread = (sched_self()->uheap != NULL) ? true : false;
```